### DISCUSSION TITLE
Collection not getting loaded, also loading progress is shown as 0, "no flush channel found for the segment, unable to flush", "Failed to get shard delegator, channel not found"

### DISCUSSION THREAD

**Discussion by maheshchil** [milvus-io/milvus#29420] [Q&A and General discussion]
Hello, 

am facing an issue with collection not getting loaded and its blocking my search operation

Below are the details related to it
Env details: 
```
- Milvus version: 2.3.3
- Deployment mode(standalone or cluster): Milvus cluster deployed in kubernetes
- MQ type(rocksmq, pulsar or kafka):   External Kafka (AWS MSK)
- SDK version(e.g. pymilvus v2.0.0rc2): Pymilvus 2.3.3
num_shards of collection: 1
Index segment size: 512MB (Default)
Vector dim: 768
Scale: 1.7 billion entities (around)
```

Other information

```
index_type: DISKANN
num of collections: 100
metric_type: L2
num of query nodes: 70 (which are of R5ad.8xlarge / R5ad.12xlarge instances types)
num of datanodes: 9
num of indexnodes: 8
num of proxy: 3
```

Schema we are using 
```
        fields = [
            FieldSchema(name='id', dtype=DataType.INT64, description="", is_primary=True, auto_id=True),
            FieldSchema(name='store_address', dtype=DataType.VARCHAR, description="", max_length=512),
            FieldSchema(name='review', dtype=DataType.VARCHAR, description="", max_length=16384),
            FieldSchema(name='vector', dtype=DataType.FLOAT_VECTOR, description="", dim=768, is_index=True),
        ]

        schema = CollectionSchema(
            fields=fields,
            description="",
            enable_dynamic_field=True
        )

        collection = Collection(name='testCollectionName', schema=schema,
                                using='default',
                                shards_num=1)
```

We are running bulk ingestion script to ingest 10k vectors in each batch upto around 1.7 billion vectors,

while am trying to load collection, am facing collection not loaded issue and also loading progress is shown as 0

Please find the milvus logs attached: milvus-log.tar.gz
[milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/13750842/milvus-log.tar.gz)

Below is an error log from Query Coordinator
`[2023/12/21 08:25:33.243 +00:00] [WARN] [task/scheduler.go:394] ["failed to promote task"] [taskID=1702983621745] [error="failed to get shard delegator: channel=by-dev-milvusdiskann1-dml_4_446340767151407526v0: channel not found"] [errorVerbose="failed to get shard delegator: channel=by-dev-milvusdiskann1-dml_4_446340767151407526v0: channel not found\n(1) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrChannelNotFound\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:490\n  | [...repeated from below...]\nWraps: (2) failed to get shard delegator\nWraps: (3) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/pkg/util/merr.wrapWithField\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:760\n  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrChannelNotFound\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:488\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).checkSegmentTaskStale\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:802\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).checkStale\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:741\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).check\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:664\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).promote\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:427\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).tryPromoteAll.func1\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:390\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskQueue).Range\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:121\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).tryPromoteAll\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:389\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).schedule\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:529\n  | github.com/milvus-io/milvus/internal/querycoordv2/task.(*taskScheduler).Dispatch\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/scheduler.go:445\n  | github.com/milvus-io/milvus/internal/querycoordv2/dist.(*distHandler).handleDistResp\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/dist/dist_handler.go:110\n  | github.com/milvus-io/milvus/internal/querycoordv2/dist.(*distHandler).start\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/dist/dist_handler.go:86\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1598\nWraps: (4) channel=by-dev-milvusdiskann1-dml_4_446340767151407526v0\nWraps: (5) channel not found\nError types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) merr.milvusError"]`

Below is a log from Data Coordinator:
```
│ milvusdiskann1-datacoord-555f76954d-st6g2 [2023/12/20 05:05:26.915 +00:00] [ERROR] [retry/retry.go:46] ["retry func failed"] ["retry time"=0] [ │
│ error="rpc error: code = Canceled desc = context canceled"] [stack="github.com/milvus-io/milvus/pkg/util/retry.Do\n\t/go/src/github.com/milvus- │
│ io/milvus/pkg/util/retry/retry.go:46\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(ClientBase[...]).call\n\t/go/src/github.com/milvus │
│ -io/milvus/internal/util/grpcclient/client.go:429\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(ClientBase[...]).Call\n\t/go/src/gith │
│ ub.com/milvus-io/milvus/internal/util/grpcclient/client.go:513\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(ClientBase[...]).ReCall\ │
│ n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:529\ngithub.com/milvus-io/milvus/internal/distributed/indexnode/clien │
│ t.wrapGrpcCall[...]\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/indexnode/client/client.go:79\ngithub.com/milvus-io/milvus/inte │
│ rnal/distributed/indexnode/client.(Client).GetJobStats\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/indexnode/client/client.go: │
│ 127\ngithub.com/milvus-io/milvus/internal/datacoord.(*IndexNodeManager).PeekClient.func1\n\t/go/src/github.com/milvus-io/milvus/internal/dataco │
│ ord/indexnode_manager.go:121"]
```

Below is log from Data Node:

```
milvusdiskann1-datanode-5cf7fc77d7-n29wc [2023/12/20 05:16:35.994 +00:00] [ERROR] [datanode/services.go:130] ["no flush channel found for the s │
│ egment, unable to flush"] [segmentID=446423094896807459] [error="failed to get flush channel has this segment: segment=446423094896807459: segm │
│ ent not found"] [errorVerbose="failed to get flush channel has this segment: segment=446423094896807459: segment not found\n(1) attached stack  │
│ trace\n -- stack trace:\n | github.com/milvus-io/milvus/pkg/util/merr.WrapErrSegmentNotFound\n | \t/go/src/github.com/milvus-io/milvus/pkg/u │
│ til/merr/utils.go:523\n | [...repeated from below...]\nWraps: (2) failed to get flush channel has this segment\nWraps: (3) attached stack trac │
```

Pls help me understand why collection is not getting loaded and this is blocking the search operation for DISKANN index types

Thanks

**Comment by maheshchil** [comment]
Below are few things which i tried

1. Increase num of query nodes to 70 (initially they were 20)
2. During the ingestion process, when the num of entities ingested was around 20 millions, collections were able to get loaded 
3. Tried checking the loading_progress after entities reached 1.7 billion, loading progress is shown as zero

![image](https://github.com/milvus-io/milvus/assets/154575954/dac8a48e-0d55-422f-a2e7-087baea13a5d)


This is the code snippet am trying to test on 1.7 billion entities, 
After collection.load() is called, its giving collection not loaded exception

![image](https://github.com/milvus-io/milvus/assets/154575954/68c3af1c-2747-4d43-bf9c-b33e350099e2)

**Comment by maheshchil** [comment]
[milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/13750844/milvus-log.tar.gz)


milvusdiskann1 (milvus cluster) logs

**Reply by yhmo** [reply]
@maheshchil 
Attach the log in the issue #29426, and close this discussion.

**Reply by maheshchil** [reply]
yes attached the logs here: https://github.com/milvus-io/milvus/issues/29426

**Comment by maheshchil** [comment]
Tracking the same discussion here: https://github.com/milvus-io/milvus/issues/29426
